### ISSUE TITLE
Expected character: ) in Regex + Regex operator expression (Invalid filter)

### ISSUE THREAD

**Issue by zhangfangkai** [json-path/JsonPath#708]
This is my JSON
{ "a": { "b": "302", "c": "test"} }

This is my JayWay JsonPath expression
[?(($.a.c =~ /^t.*/) && ($a.b =~ /^3\\d{2}/))]

I'm getting an error Expected character: ) when running here

Any reason why ?

**Comment by tomakehurst** [comment]
I'm seeing something similar also - seems that the parser stopped being able to handle parentheses inside regex expressions between 2.4.0 and 2.5.0.

I get a parse error in 2.5.0 for this expression, but it worked in 2.4.0:
`$[?(@.['name'] =~ /([a-zA-Z0-9]+)?/)]`

**Comment by zhangfangkai** [comment]
Thanks, when i use the 2.4.0 version, the error disappear, it seems like a bug when update from 2.4.0 to 2.5.0

**Comment by Youngea** [comment]
Hey, I'm interested in looking into the problem and fixing the issue

**Comment by hezonghan** [comment]
Hello, I tested your issue with the following code:
```java
        String json_str = "[" +
                "{ \"name\": \"_ad\", \"id\": 1 }" +
                "{ \"name\": \"dd234\", \"id\": 2 }" +
                "{ \"name\": \"99_s\", \"id\": 3 }" +
                "{ \"name\": \"qwerty\", \"id\": 4 }" +
                "]";
        String path_str = "$[?(@.['name'] =~ /([a-zA-Z0-9]+)?/)]";  //  ([a-zA-Z0-9]+)?

        List<Object> list = JsonPath.read(json_str , path_str);
        System.out.println(list);
```
using the code forked on yesterday afternoon.
The output is:
```
[{"name":"dd234","id":2},{"name":"qwerty","id":4}]
```
It seems that there is no problem in current code? Someone may have fixed it.

**Comment by moshevi** [comment]
@zhangfangkai That was fixed in https://github.com/json-path/JsonPath/commit/e68e9f2f9b04d741b85797335ac78f1a8c7317c2, wasn't released yet.

**Comment by kallestenflo** [comment]
Fixed in 2.6.0
